packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -295,22 +295,43 @@ class RenderEditable extends RenderBox {
   ValueListenable<bool> get selectionEndInViewport => _selectionEndInViewport;
   final ValueNotifier<bool> _selectionEndInViewport = ValueNotifier<bool>(true);
 
+  // Returns a bool indicating if the offset is within factor of being contained
+  // within the rect.
+  static bool _containsFuzzy(Rect rect, Offset offset, [double factor = 0.1]) {
```

Comment:
Any better way to do this?

Reply:
Instead of a fuzzy check, it may be better to instead increase the size of the clipping zone by using the `TextPainter.preferredLineHeight`. This will do a simple layout and size it as expected. It may be the case that we are already doing this based off of some measured metric.

Another thing that could cause this is that since we now enforce strut on TextFields, it is now possible for text to extend past the provided space due to different fonts being fallen back on.

Reply:
Overall, it does seem like a fuzzy check is a hack and adds slop to the way everything works, I'd prefer a more formal and rigid solution if it is achievable. Feel free to come chat about other approaches to this.